shuup_tests/gdpr/test_user_creation.py

```diff
@@ -0,0 +1,135 @@
+            "product_id": inputs["product_id"],
+            "quantity": 1,
+            "supplier": get_default_supplier().pk,
+        }
+        add_to_basket_resp = client.post(basket_path, data=basket_data)
+        assert add_to_basket_resp.status_code < 400
+        product_ids.append(inputs["product_id"])
+    basket_soup = client.soup(basket_path)
+    assert b'no such element' not in basket_soup.renderContents(), 'All product details are not rendered correctly'
+    return product_ids
+
+@pytest.mark.parametrize("gdpr_enabled, create_contact", [
+    (True, True),
+    (False, True),
+    (True, False),
+    (False, False),
+])
+def test_user_creation_in_order_flow(regular_user, gdpr_enabled, create_contact):
+    cache.clear()
+    create_default_order_statuses()
+    n_orders_pre = Order.objects.count()
+    populate_if_required()
+    if create_contact:
+        c, shop, contact = initialize_client_and_parameters(regular_user)
+    else:
+        c = SmartClient()
+        shop = get_default_shop()
+    product_ids = _populate_client_basket(c)
+
+    if gdpr_enabled:
+        gdpr_settings = GDPRSettings.get_for_shop(shop)
+        gdpr_settings.enabled = True
+        gdpr_settings.save()
+    else:
+        gdpr_settings = GDPRSettings.get_for_shop(shop)
+        gdpr_settings.enabled = False
+        gdpr_settings.save()
+    addresses_path = reverse("shuup:checkout", kwargs={"phase": "addresses"})
+    addresses_soup = c.soup(addresses_path)
+    inputs = fill_address_inputs(addresses_soup)
+    response = c.post(addresses_path, data=inputs)
+    assert response.status_code == 302  # Should redirect forth
+
+    # Make sure the address is initialized from storage
+    # Go back to addresses right before back to methods
+    c.soup(addresses_path)
+
+    methods_path = reverse("shuup:checkout", kwargs={"phase": "methods"})
+    methods_soup = c.soup(methods_path)
+    assert c.post(methods_path, data=extract_form_fields(methods_soup)).status_code == 302  # Should redirect forth
+
+    checkout_complete.connect(checkout_complete_signal, dispatch_uid="checkout_complete_signal")
+
+    confirm_path = reverse("shuup:checkout", kwargs={"phase": "confirm"})
+    confirm_soup = c.soup(confirm_path)
+    Product.objects.get(pk=product_ids[0]).soft_delete()
+    assert c.post(confirm_path, data=extract_form_fields(confirm_soup)).status_code == 200  # user needs to reconfirm
+    data = extract_form_fields(confirm_soup)
+    data['product_ids'] = ','.join(product_ids[1:])
+    assert c.post(confirm_path, data=data).status_code == 302  # Should redirect forth
+
+    n_orders_post = Order.objects.count()
+    assert n_orders_post > n_orders_pre, "order was created"
+
+    order = Order.objects.first()
+    expected_ip = "127.0.0.2"
+    assert order.ip_address == expected_ip
+    if create_contact:
+        assert order.orderer == contact
+    elif gdpr_enabled:
+        assert type(order.orderer) == PersonContact
```

Comment:
You could check the user and customer fields here as well and also check the email/username matches to what we expect.